utility_analysis_new/combiners.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2022 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""UtilityAnalysisCountCombiner."""
+
+from dataclasses import dataclass
+from typing import List, Sized, Tuple
+import numpy as np
+
+import pipeline_dp
+from pipeline_dp import dp_computations
+
+
+@dataclass
+class CountUtilityAnalysisMetrics:
+    """ Stores metrics for the count utility analysis.
+
+    Args:
+        count: actual count of contributions per partition.
+        per_partition_contribution_error: the amount of contribution error in a partition.
+        expected_cross_partition_error: the estimated amount of error across partitions.
+        std_cross_partition_error: the standard deviation of the contribution bounding error.
+        std_noise: the noise standard deviation for DP count.
+        noise_kind: the type of noise used.
+    """
+    count: int
+    per_partition_contribution_error: int
+    expected_cross_partition_error: float
+    std_cross_partition_error: float
+    std_noise: float
+    noise_kind: pipeline_dp.NoiseKind
+
+
+@dataclass
+class UtilityAnalysisCountAccumulator:
+    count: int
+    per_partition_contribution_error: int
+    expected_cross_partition_error: float
+    var_cross_partition_error: float
+
+
+class UtilityAnalysisCountCombiner(pipeline_dp.Combiner):
+    """ A combiner for utility analysis counts. """
```

Comment:
Nit: """A ...counts.""" (no spaces between """ and text and in other places)

Reply:
Fixed here and in other places, apologies for my auto-format. I've fixed it!